### PR TITLE
Allow annotations for the ingress at the app level

### PR DESCRIPTION
**Pull request by annymsMthd**
Signed-off-by: Joshua Benjamin <joshuabnjmn@gmail.com>

### CODE REVIEW

**Comment by annymsMthd** [comment]
Our k8s clusters have multiple ingress controllers so it would be helpful to allow annotations for ingresses:
```
ingressConfig:
    annotations:
      kubernetes.io/ingress.class: "private"
```

File: pkg/apis/app/v1alpha1/types.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type FlinkApplicationSpec struct {
 	FlinkVersion      string                       `json:"flinkVersion"`
 	TaskManagerConfig TaskManagerConfig            `json:"taskManagerConfig,omitempty"`
 	JobManagerConfig  JobManagerConfig             `json:"jobManagerConfig,omitempty"`
+	IngressConfig     IngressConfig                `json:"ingressConfig,omitempty"`
```
Comment (anandswaminathan) [inline pkg/apis/app/v1alpha1/types.go]: @annymsMthd 

Right now for an application when an Ingress is created, it is never updated (unlike Service and Deployment). Let's say you add an annotation for Ingress config, and you update it, the update will never be reflected in the Ingress.

Comment (annymsMthd) [reply pkg/apis/app/v1alpha1/types.go]: Okay. I can add that if you guys like this addition to the api.

Comment (anandswaminathan) [reply pkg/apis/app/v1alpha1/types.go]: Will have to think it through. We currently do not have this requirement.

We pass on the annotation from the flink application to the deployments. Would doing something similar for ingress and services address your use case?

Comment (annymsMthd) [reply pkg/apis/app/v1alpha1/types.go]: That would work.

Comment (anandswaminathan) [reply pkg/apis/app/v1alpha1/types.go]: Sorry for the delay. Forgot about this thread.

Submitted a PR: https://github.com/lyft/flinkk8soperator/pull/74

**Pull request closed without merging**